packages/api/src/routes/tasks.ts

```diff
@@ -29,9 +43,8 @@ router.get("/custom/:id", (req, res, next) => {
     .catch(next);
 });
 
-router.post("/:id/custom", (req, res, next) => {
+router.post("/:id/custom", isAuthenticated, (req, res, next) => {
   const studyId = req.query.studyId;
```

Comment:
I couldn't figure out a way to simplify this endpoint. I tried getting the study with Study.findById but I can't access its batteries because it returns type Query

Reply:
oh just saw this, you just have to `await` Querys to get the data